fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1953,7 +1987,160 @@ public IndexingPolicy build() {
+        }
+
+        /**
+         * Builder for {@link ScrubbingPolicy}.
+         *
+         * <pre><code>
+         * OnlineIndexer.ScrubbingPolicy.newBuilder().setReportMissingLimit(100).setScrubDangling(false).build()
+         * </code></pre>
+         *
+         */
+        @API(API.Status.UNSTABLE)
+        public static class Builder {
+            boolean scrubDangling = true;
+            boolean scrubMissing = true;
+            int reportDanglingLimit = 1000;
+            int reportMissingLimit = 1000;
+            boolean allowRepair = true;
+            long entriesScanLimit = 0;
+
+            protected Builder() {
+            }
+
+            /**
+             * If never called, the defaulted to allow scanning the index, looking for dangling index entries.
+             * @param scrubDangling if true
+             * @return this builder.
+             */
+            public Builder setScrubDangling(final boolean scrubDangling) {
+                this.scrubDangling = scrubDangling;
+                return this;
+            }
+
+            /**
+             * If never called, the defaulted to allow scanning the record store, looking for missing index entries.
+             * @param scrubMissing if true
+             * @return this builder.
+             */
+            public Builder setScrubMissing(final boolean scrubMissing) {
+                this.scrubMissing = scrubMissing;
+                return this;
+            }
+
+            /**
+             * If never called, the default is allowing (up to) 1000 error reports.
+             * @param reportDanglingLimit the max number of dangling-indexes errors to report.
+             * @return this builder.
+             */
+            public Builder setReportDanglingLimit(final int reportDanglingLimit) {
+                this.reportDanglingLimit = reportDanglingLimit;
+                return this;
+            }
+
+            /**
+             * If never called, the default is allowing (up to) 1000 error reports.
+             * @param reportMissingLimit the max number of missing-indexes errors to report.
+             * @return this builder.
+             */
+            public Builder setReportMissingLimit(final int reportMissingLimit) {
+                this.reportMissingLimit = reportMissingLimit;
+                return this;
+            }
+
+            /**
+             * Set whether the scrubber, if finding an error, will repair it.
+             * @param val - if false, always report errors but do not repair.
+             *            - if true (default), report errors (up to report limits) but always repair.
+             * @return this builder.
+             */
+            public Builder setAllowRepair(boolean val) {
+                this.allowRepair = val;
+                return this;
+            }
+
+            /**
+             * Set records/index entries scan limit. The scrubbing task will return after scanning more than this limit.
```

Comment:
You are right, this is not a real limit, but more of a quota. 
The indexer/scrubber reads chunks of entries, while the throttle increases/decreases the chunks' size (of to config's max limit). The scrubber stops when the number of scanned entries exceeds the limit/quota. How should it be called?

Reply:
I guess the main question I'd like answered is some clarity around how much it can exceed the quota/limit. It's the "more than" language that's tripping me up

Reply:
That's hard to tell as the caller can overwrite the default max chunk limit (100 items) with its own.

Reply:
Hm, well, ideally it would let the caller know that the scan limit may be exceeded by up to the chunk limit or something along those lines (or the scrubber would strictly enforce the scan limit to make the API more understandable).